dips/dip-158.md

```diff
@@ -0,0 +1,795 @@
+
+
+### **Auth / Capture Payment Type**
+Auth / Capture payment, as the name suggests, means that the payment is divided into two steps:
+* Authorization
+* Capture
+ 
+The Authorization step ends when the funds are locked by the sender VASP so they can later be captured by the receiver VASP. Note that the capture may occur several days after a successful authorization. It is up to the receiver VASP to decide on the timing. In addition, the receiver may choose to capture only a partial amount rather than the full amount which was authorized. 
+
+This payment type is useful for delayed fulfillment or pre-authorizing an expected amount to ensure that an amount can be captured after services are rendered. 
+
+Off-chain negotiations for Auth / Capture payment are divided into two steps accordingly. First, off-chain negotiations will result in funds being authorized, i.e. locked by the wallet. Later, a separate off-chain negotiation triggered by the receiving VASP will capture the funds. Note that for the time being, multi-capture is not supported and there can be only a single capture request.
+
+Successful authorization will not necessarily be followed by a successful capture. It is possible that the receiver will choose to cancel the authorization and essentially void the payment before it was captured. One possible reason to void a payment is that the products for which an amount was locked are out of stock and cannot be fulfilled. 
+
+---
+## New Elements and Existing Elements Extensions
+---
+This section specifies extensions for existing elements (e.g. new values) and introduces new ones required for P2M scenarios.
+
+
+## PaymentActorObject Extension
+The PaymentActorObject is extended to include the field `merchant_data` of type `MerchantDataObject`.
+Field|Type|Required?|Description|
+|-|-|-|-|
+|merchant_data|MerchantDataObject|N|This element is required for some scenarios. It includes the details of the merchant to be displayed to the payer|
+|
+
+## PaymentActor Status Enum Extension
+The following values are added to the status enum of [PaymentActor](https://dip.diem.com/dip-1/#paymentactorobject) object:
+
+* `authorized` - This means the actor has performed all required tasks and funds are locked for a future capture. Note that the actual lock is done by the wallet (sender), so for the receiver this status means that all risk checks were successful and it is willing to accept the payment. 
+* `needs_init_data` - This means the actor requires basic data to initialize the payment. For the receiver, this status means that the payer data is required. For the sender, it means that the merchant data is required  
+
+For completeness purposes, the full list of values is: 
+* `none`
+* `needs_kyc_data`
+* `ready_for_settlement`
+* `abort`
+* `authorized`
+* `soft_match`
+* `needs_init_data`
+
```

Comment:
Let's discuss this.
The new statuses (authorized and needs_init_data) are required for P2M and are used in the new flow of messages (not as part of a shared object).
The statuses soft_match and needs_kyc_data are not required for P2M. They are only listed in the DIP for readers' convenience and completeness, i.e. so all possible values of the enum will be specified in a single DIP rather than being spread over several.

The transition table in this section specifies which statuses and transitions are relevant to P2M.